infrastructure/src/main/java/org/corfudb/infrastructure/ServerContext.java

```diff
@@ -683,6 +685,42 @@ public Layout copyManagementLayout() {
         }
     }
 
+    /**
+     * Shuts down the workerGroup and spawns a new one
+     */
+    public void refreshWorkerGroupThreads() {
+        CorfuRuntime.CorfuRuntimeParameters params = getManagementRuntimeParameters();
+
+        workerGroup.shutdownGracefully(
+                params.getNettyShutdownQuitePeriod(),
+                params.getNettyShutdownTimeout(),
+                TimeUnit.MILLISECONDS
+        );
+
+        try {
+            // Wait a while for existing tasks to terminate
+            if (!workerGroup.awaitTermination(params.getNettyShutdownTimeout(), TimeUnit.MILLISECONDS)) {
+                log.error("refreshWorkerGroup: Executor Pool workerGroup did not terminate.");
+            } else {
+                log.info("refreshWorkerGroup: Executor Pool workerGroup terminated successfully.");
+            }
+        } catch (InterruptedException ie) {
+            // (Re-)Cancel if current thread also interrupted
+            workerGroup.shutdownNow();
+            // Preserve interrupt status on the current thread
+            Thread.currentThread().interrupt();
+        }
+
+        log.trace("refreshWorkerGroup: workerGroup isShuttingDown {}," +
```

Comment:
Since this log line provides strictly more information than the one below, can we mark this one as an info level log and make the other a trace level log instead (or simply remove it altogether)?

Reply:
Merged them now, and changed to `info`. Thanks!